doc/ref_arch/kubernetes/chapters/chapter01.md

```diff
@@ -54,7 +120,7 @@ Figure 1-1 below shows this at a high level.
 <p align="center"><img src="../figures/ch01_k8s_arch.png" alt="Kubernetes Architecture for NFV" title="Kubernetes Architecture for NFV" width="100%"/></p>
 <p align="center"><b>Figure 1-1:</b> Kubernetes Architecture for NFV</p>
 
-### Other thoughts
+### More thoughts
 - Kubernetes / CaaS should be recognised in ETSI NFV / MANO
 - It is feasible that Kubernetes, using operators and custom resources, could become a generic VNFM (i.e. the Kubernetes parts are the underlying engine, with the operators and custom resources being the equivalent of the specific VNFM).
```

Comment:
you mean VIM? :)

Reply:
No - I mean VNFM :)

Reply:
I don't see K8s becoming the new VNFM, TBH it rather reduces the need to have VNFM in general. The key is to make it mandatory to rely on K8s for the LCM of the CNF in terms of upgrade, healing, etc. Then operators kind-of become a neccesity for stateful parts (databases). But the point with K8s is that the VNFM is not needed, or even if it's needed, a much thinner VNFM is necessary.

Reply:
Where would you position Helm in this environment?  Mandated? Optional?  Would you consider it taking on the role of a lighter-weight VNFM?

Reply:
> Where would you position Helm in this environment? Mandated? Optional? Would you consider it taking on the role of a lighter-weight VNFM?

@kedmison, I will go with it should be part of the CNF ( VNF) package onboarding task, so either from VNF-M or NFVO

Reply:
@ASawwaf in my view it is more the capability of the VIM to manage complex applications and not part of the VNFM or the NFVO.

Reply:
> 
> 
> > Where would you position Helm in this environment? Mandated? Optional? Would you consider it taking on the role of a lighter-weight VNFM?
> 
> @kedmison, I will go with it should be part of the CNF ( VNF) package onboarding task, so either from VNF-M or NFVO

Part of the VIM (CISM is the proper term I believe). Also part of the CaaS (there is a need for a package manager to allow an abstraction that corresponds to "an application").

Reply:
> Where would you position Helm in this environment? Mandated? Optional? Would you consider it taking on the role of a lighter-weight VNFM?

Arguably, Helm shouldn't be positioned at all here. How to do package management is an implementation detail. It's as (non-)important as discussing "should we agree on apt + deb-packages or yum + rpm-packages".

It will also evolve over time. Helm was last year's darling, this year it's kustomizations and operators, next year something else.

Reply:
isn't CNTT ultimately for nailing down implementation details?
BTW can you show me a CaaS infra which does not currently include Helm?

ok, let's not say Helm explicitly, but I would argue a "package manager" is a must have CaaS component

Reply:
I agree re: nailing down implementation details.  Sometimes that is at RA level, and sometimes at RI level.  I think we have to specify the package manager as part of the RA; CNF developers need to know how to package their software in order to successfully deliver to Implementations that meet this Architecture.

Reply:
I strongly disagree. A golden rule of standardization (and this is a standardization exercise at the end of the day!) is to specify external behavior and explicitly _not_ prescribe implementations (and to know where _not_ to standardize). The reason is that you typically don't gain anything with respect to interoperability (because you're addressing a symptom, not a cause) but you risk stifling innovation and competition.

Reply:
do you disagree with explicitly naming a package manager implementation in the RA, or with the statement that package management is required in CaaS?

Reply:
> BTW can you show me a CaaS infra which does not currently include Helm?

That's a trick question :)
All k8s distros _support_ Helm, because it runs on any k8s.
Few _include_ it (e.g. OpenShift, GKE, AKS, Weaveworks, etc. don't), because it's trivial to install on top if someone wants it.
None of them would make itself _dependent_ on it.

The more important question is which tool has the most momentum with the app developer and operations communities. And this is where Helm lost momentum versus other tools in my observation. For example:
* "kustomize" seems popular with those working with a devops (esp. gitops) methodology, are deploying mainly stateless, componentized "web service" kind of workloads, and don't want/need a package manager in the loop. That Google integrated it into kubectl will likely make it the de facto standard in that space.
* "operators" seem popular with those working with complex, often stateful workloads that are very common in the enterprise/telco space. These workloads are notoriously hard to 
operate and what you want is to bottle/"package" this operational knowledge somehow. This is where Helm's package management approach has little to offer, forcing Chart developers to use awkward, hard to maintain workarounds.

Reply:
> do you disagree with explicitly naming a package manager implementation in the RA, or with the statement that package management is required in CaaS?

I wouldn't say a package manager is _required_ and in fact the kustomize model (result of the deep analysis of app config management in [1]) shows how one can do pretty well without.
But my disagreement was with picking a single PM tool and declaring it "the standard" by adding it to the RA, especially when done without analysis of alternatives and when it might lead to a disconnect between telco and the direction of the wider community.

[1] https://docs.google.com/document/d/1cLPGweVEYrVqQvBLJg6sxV-TrE5Rm2MNOBA_cxZP2WU/edit

Reply:
That's a great document, and I'll have to go through it a few times to really internalize all the content.  My high-level reactions to it are:
1) we need to define the package management pattern for CNFs to use, whether it be this model, Helm, or something else, if only so that CNF vendors know what to build, test, and expect from K8S clusters, and so that we can build automation pipelines, from CNF vendor through to production in operator networks, that are well tested and work consistently and reliably.
2) Our selection process should be weighted both on what is popular in the industry (i.e. has a well-supported ecosystem around it) as well as its intrinsic technical merits.

Reply:
I have updated the terminology for CaaS, removed PaaS, but haven't added package management as a constituent part of the CaaS.